core/src/main/java/org/apache/calcite/rel/rules/AggregateReduceFunctionsRule.java

```diff
@@ -97,16 +100,66 @@
       new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()),
           RelFactories.LOGICAL_BUILDER);
 
+  private final Set<SqlKind> functionsToReduce;
+
+  /**
+   * Gets an instance of AggregateReduceFunctionsRule
+   * with client provided specific functions to reduce
+   * @param functionsToReduce client provided information
+   *                          on which specific functions will be
+   *                          reduced by this rule
+   * @return an instance of AggregateReduceFunctionsRule
+   */
+  public static AggregateReduceFunctionsRule
+      getInstanceWithSpecificFunctionsToReduce(final Set<SqlKind> functionsToReduce) {
+    Preconditions.checkArgument(functionsToReduce != null,
+        "Error: expecting a valid handle for AggregateFunctionsToReduce");
+    return new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()),
+      RelFactories.LOGICAL_BUILDER, functionsToReduce);
+  }
+
   //~ Constructors -----------------------------------------------------------
 
   /** Creates an AggregateReduceFunctionsRule. */
   public AggregateReduceFunctionsRule(RelOptRuleOperand operand,
       RelBuilderFactory relBuilderFactory) {
     super(operand, relBuilderFactory, null);
+    // by default, this rule will reduce all functions it handles
+    this.functionsToReduce = new HashSet<>();
+    addDefaultSetOfFunctionsToReduce();
+  }
+
+  /**
+   * Creates an AggregateReduceFunctionsRule with client
+   * provided information on which specific functions will
+   * be reduced by this rule
+   * @param operand root operand
+   * @param relBuilderFactory builder for relational expressions
+   * @param functionsToReduce client provided information
+   *                          on which specific functions
+   *                          will be reduced by this rule
+   */
+  public AggregateReduceFunctionsRule(RelOptRuleOperand operand,
```

Comment:
Providing a public constuctor with operand directly is not a very good idea. Plus it is not correct since the rule expects to work only with Aggregate. I would prefer the pattern used in ProjectFilterTransposeRule for instance where the constructor takes as an argument the class of the matching operand (i.e., `Class<? extends Aggregate> aggregateClass`). This should make also the constructor slighly easier to use.

Reply:
Done.